framework/core/render_pass.cpp

```diff
@@ -244,23 +244,73 @@ void set_attachment_layouts(std::vector<T_SubpassDescription> &subpass_descripti
 	}
 }
 
+/**
+ * @brief Assuming there is only one depth attachment
+ */
+template <typename T_SubpassDescription, typename T_AttachmentDescription>
+bool is_depth_a_dependency(std::vector<T_SubpassDescription> &subpass_descriptions, std::vector<T_AttachmentDescription> &attachment_descriptions)
+{
+	uint32_t times_used{0};
+	for (auto &subpass : subpass_descriptions)
+	{
+		if (subpass.pDepthStencilAttachment)
+		{
+			times_used++;
+			if (times_used > 1)
+			{
+				return true;
+			}
+		}
+	}
+
+	for (auto &subpass : subpass_descriptions)
+	{
+		for (size_t k = 0U; k < subpass.inputAttachmentCount; ++k)
+		{
+			const auto &reference = subpass.pInputAttachments[k];
+
+			if (vkb::is_depth_format(attachment_descriptions[reference.attachment].format))
+			{
+				// Depth used as input
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
 template <typename T>
-std::vector<T> get_subpass_dependencies(const size_t subpass_count)
+std::vector<T> get_subpass_dependencies(const size_t subpass_count, bool depth_stencil_dependency = false)
```

Comment:
The default argument for `depth_stencil_dependency` is not needed. This function is always (one time!) called with full argument set.

Reply:
Done